doc/smart-switch/PA-Validation/SmartSwitchPAValidationOffload.md

```diff
@@ -0,0 +1,172 @@
+
+## 2.2 PA Validation Offload
+
+A new orchagent (PaValidationOffloadOrch) is responsible for offloading the PA_VALIDATION table entries. Once started by DashOffloadManager, it subscribes to the PA_VALIDATION table and maps the PA validation entries into the ACL configuration.
+
+### 2.2.1 PA Validation Offload ACL configuration
+
+The offloading is achieved by using a User-Defined EGRESS ACL table bound to the NPU<->DPU port. The table is created/managed per DPU.
+
+```
+{
+    "ACL_TABLE_TYPE": {
+        "DASH_PA_VALIDATION": {
+            "MATCHES": [
+                "TUNNEL_VNI",
+                "SRC_IP",
+                "SRC_IPV6"
+            ],
+            "ACTIONS": [
+                "PACKET_ACTION"
+            ],
+            "BIND_POINTS": [
+                "PORT"
+            ]
+        }
+    }
+}
+```
+
+Please refer to https://github.com/sonic-net/SONiC/blob/master/doc/acl/ACL-Table-Type-HLD.md for more details regarding User-Defined ACL tables.
+
+The ACL table configuration:
+
+```
+{
+  "ACL_TABLE": {
+    "DASH_PA_VALIDATION_DPU0": {
+      "STAGE": "EGRESS",
+      "TYPE": "DASH_PA_VALIDATION",
+      "PORTS": ["Ethernet0"]
+      }
+    }
+}
+```
+
+Each PA Validation entry is translated into the following set of rules:
+
+1) For every address in pa_validation.addresses:
+```
+{
+  "ACL_RULE": {
+    "DASH_PA_VALIDATION_DPU0|RULE_VNI_{pa_validation.vni}_{idx}": {
+      "PRIORITY": "10",
+      "PACKET_ACTION": "ACCEPT",
+      "SRC_IP": "{address}/32",
+      "TUNNEL_VNI": {pa_validation.vni}
+    }
+  }
+}
+```
+
+2) Single drop rule with the lower priority:
+```
+{
+  "ACL_RULE": {
+    "DASH_PA_VALIDATION_DPU0|RULE_VNI_{pa_validation.vni}_DROP": {
+      "PRIORITY": "1",
+      "PACKET_ACTION": "DROP",
+      "TUNNEL_VNI": {pa_validation.vni}
+    }
+  }
+}
+```
+
+<img src="images/DashOffloadAcl.svg">
+
+### 2.2.1 PA Validation Offload GNMI feedback
+To preserve the GNMI feedback behavior, the offload logic must also create an entry in the DPU's APPL_STATE_DB.
+For each PA validation processed, the PaValidationOffloadOrch creates the following entry:
+
+```
+DASH_PA_VALIDATION_TABLE:{{vni}}
+    "result": {{result}}
+```
+
+The result is 0 for success, or > 0 for error code
+
+Please refer to https://github.com/sonic-net/SONiC/pull/1759 for more details regarding GNMI feedback requirements and behavior.
+
+## 2.3 DPU Shut/Restart
+When DPU goes down/restarts, the ACL configuration should be cleaned. It's done by the Dash Offload Manager which listens to the ChassisStateDB DPU_STATE Table. When it detects that the DPU is down (dpu_control_plane_state is down), the PaValidationOffloadOrch is deinitialized, leading to ACL configuration cleanup and ZMQ proxy subscription removal.

```

Comment:
we might also need to cover the DPU upgrade case, because when chassis db is updated, this orch might not be running at all.

@ganglyu  to confirm.

Reply:
Theoretically, we need to upgrade this orch when we upgrade the DPU.